### ISSUE TITLE
Change from `tuple` to `struct` for storing metadata of stored collections

### ISSUE THREAD

**Issue by tmadlener** [AIDASoft/podio#705]
Currently we use a `std::tuple` to store some metatdata related to collections:

https://github.com/AIDASoft/podio/blob/5afdefdd233bd6b53f51d38bceffae600ae41409/include/podio/utilities/RootHelpers.h#L18-L21

This makes it hard to read via julia and also from a maintainability point of view it would be nicer, if this was something like

```cpp
struct CollectionWriteInfo {
  uint32_t collectionID;
  std::string dataType;
  bool isSubset;
  unsigned int schemaVersion;
};
```

@peremato I suppose something along these lines should work?

**Comment by peremato** [comment]
What about adding also the collection name? In this case you need only one branch for all information.

**Comment by tmadlener** [comment]
Yeah, I could add that. I have to check how we can deduplicate some parts, because that information is also stored via the CollectionIDTable.
